### ISSUE TITLE
谁成功在jetson上使用了 flash_attn

### ISSUE THREAD

**Issue by cthulhu-tww** [Dao-AILab/flash-attention#986]
我刚才在我jetson上成功编译了这个库，但是运行llm推理的时候，报错
RuntimeError: CUDA error: no kernel image is available for execution on the device
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

jetpack：6
cuda：12.2
python：3.10
torch：用的英伟达的轮子[torch-2.3.0a0+6ddf5cf85e.nv24.04.14026654-cp310-cp310-linux_aarch64.whl](https://developer.download.nvidia.com/compute/redist/jp/v60dp/pytorch/torch-2.3.0a0+6ddf5cf85e.nv24.04.14026654-cp310-cp310-linux_aarch64.whl)

**Comment by mikeshi80** [comment]
Yes, I did. If you still want to know solution, you can leave a message. Maybe I will write a blog sometime later.

**Comment by Gear-dev-sudo** [comment]
> Yes, I did. If you still want to know solution, you can leave a message. Maybe I will write a blog sometime later.

Yes, please, I ran into the same issue

**Comment by mikeshi80** [comment]
> > Yes, I did. If you still want to know solution, you can leave a message. Maybe I will write a blog sometime later.
> 
> Yes, please, I ran into the same issue

OK, I will do it ASAP, however, since it is written in Chinese, maybe you need use translator to read it.

**Comment by Gear-dev-sudo** [comment]
> > > Yes, I did. If you still want to know solution, you can leave a message. Maybe I will write a blog sometime later.
> > 
> > 
> > Yes, please, I ran into the same issue
> 
> OK, I will do it ASAP, however, since it is written in Chinese, maybe you need use translator to read it.
> 
> PS: OK, I read your bio. Obviously, you can read Chinese

Thank you, I am Chinese myself haha, I'll try to translate your blog into English for others to reference

**Comment by thohemp** [comment]
Having the same issue.

**Comment by mikeshi80** [comment]
I wrote the approach here [install vllm on jetson](https://www.mikeshi.me/posts/vllm-on-jetson/), and there is a chapter about how to install vllm-flash-attn on Jetson. Although the name is different, but vllm-flash-attn is just the fork of flash-attn, they share almost all source code.

**Comment by mikeshi80** [comment]
The blog article is written in Chinese, translation is welcomed.

**Comment by linhuaiyi** [comment]
> The blog article is written in Chinese, translation is welcomed.

I note that there is no info about jetpack,cuda,python version. 
Would you mind supply these info. 
Cuz I found a lot of problems especailly when compile things about cuda according to your blog.

**Comment by Xiong-Jirong** [comment]
> JetPack 6, with the built-in CUDA 12.2. And the torch is 2.3.0, you need build it yourself.

您好，请问您博客中的方法，适用于jetpack6.1 CUDA12.6吗

**Comment by mikeshi80** [comment]
> > JetPack 6, with the built-in CUDA 12.2. And the torch is 2.3.0, you need build it yourself.
> 
> 您好，请问您博客中的方法，适用于jetpack6.1 CUDA12.6吗

Sorry，这是我借用公司的硬件做的，现在它已经不在我手头了，等我有机会再试试

**Comment by icm-ai** [comment]
@

> > JetPack 6, with the built-in CUDA 12.2. And the torch is 2.3.0, you need build it yourself.
> 
> 您好，请问您博客中的方法，适用于jetpack6.1 CUDA12.6吗

我也正在尝试Jetpack6.1/6.2 CUDA12.6，用的硬件是Jetson orin nx 16G

**Comment by Sun-Xiaohui** [comment]
@icm-ai  你好，请问你能成功安装flash_attn吗，我在jetson orin nx上尝试多次都失败了，是否有预编译好的whl？

**Comment by icm-ai** [comment]
> [@icm-ai](https://github.com/icm-ai) 你好，请问你能成功安装flash_attn吗，我在jetson orin nx上尝试多次都失败了，是否有预编译好的whl？

我编译能通过，但是使用的过程中会报错。
后面转为使用 [jetson-container](https://github.com/dusty-nv/jetson-containers/tree/master/packages/attention/flash-attention) 了，这个方便，可以避免环境的问题，建议你也尝试这种方式。

**Comment by icm-ai** [comment]
> [@icm-ai](https://github.com/icm-ai) 你好，请问你能成功安装flash_attn吗，我在jetson orin nx上尝试多次都失败了，是否有预编译好的whl？

[flash-attention](https://github.com/dusty-nv/jetson-containers/tree/master/packages/attention/flash-attention)
